<?xml version="1.0"?>
<launch>
  <param name="/processes/sample_detection/search/frustum_buffer" type="double" value="200"/>

  <include file="$(find saliency_detector)/launch/saliency_detector.launch">
    <arg name="namespace" value="port"/>
  </include>

  <include file="$(find saliency_detector)/launch/saliency_detector.launch">
    <arg name="namespace" value="center"/>
  </include>

  <include file="$(find saliency_detector)/launch/saliency_detector.launch">
    <arg name="namespace" value="starboard"/>
  </include>

  <node
    pkg="detection_filter"
    type="dslr_kalman_filter_node"
    name="kalman_filter_node"
    machine="search"
    ns="/processes/sample_detection/search">
    <remap from="ack" to="/processes/executive/pursuit_result"/>
    <remap from="odometry" to="/motion/odometry"/>
    <remap from="pause_state" to="/io/pause_state"/>
    <param name="max_dist" type="double" value="0.2"/>
    <param name="max_cov" type="double" value="1.7"/>
    <param name="max_pub_cov" type="double" value="0.01"/>
    <param name="max_pub_vel" type="double" value="0.1"/>
    <param name="positive_exclusion_radius" type="double" value="10.0"/>
    <param name="negative_exclusion_radius" type="double" value="0.2"/>
    <param name="error_cov_post" type="double" value="0.09"/>
    <param name="process_noise_cov" type="double" value="0.00025"/>
    <param name="measurement_noise_cov" type="double" value="0.01"/>
    <param name="period" type="double" value="0.33"/>
    <param name="PDgO" type="double" value="0.95"/> <!-- true positive rate -->
    <param name="PDgo" type="double" value="0.10"/> <!-- false positive rate -->
    <param name="PO_init" type="double" value="0.01"/>
    <param name="pub_certainty_thresh" type="double" value="0.9"/>
    <param name="min_certainty" type="double" value="0.01"/>
    <param name="odometry_tick_dist" type="double" value="5.0"/>
    <param name="exclusion_zone_range" type="double" value="50.0"/>
    <param name="max_colormodel_distance" type="double" value="0.2"/>
    <param name="positive_exclusion_radius" type="double" value="10.0"/>
    <param name="negative_exclusion_radius" type="double" value="0.2"/>
    <remap from="frustum_buffer" to="/processes/sample_detection/search/frustum_buffer"/>
  </node>

  <!--
  <node
    pkg="sample_detection"
    type="ray_to_points.py"
    name="ray_to_points"
    machine="search"
    ns="/processes/sample_detection/search">
    <remap from="fence_line" to="/processes/fence_detection/fence_line"/>
  </node>

  <node
    pkg="saliency_detector"
    type="close_range_detector.py"
    name="close_range_detector"
    machine="search"
    ns="/processes/sample_detection/search">
    <remap from="cam_info" to="/cameras/search/center/camera_info"/>
    <remap from="image" to="/cameras/search/center/image_color"/>
  </node>
  -->

</launch>
